tests/test_cpuid.sh

```diff
@@ -0,0 +1,124 @@
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+# ------------------------------------------------------------------------------
+# Colors
+# ------------------------------------------------------------------------------
+
+CB='\033[1;35m'
+CC='\033[1;36m'
+CE='\033[0m'
+
+# ------------------------------------------------------------------------------
+# Environment
+# ------------------------------------------------------------------------------
+
+NUM_CORES=$(grep -c ^processor /proc/cpuinfo)
+
+# ------------------------------------------------------------------------------
+# Helpers
+# ------------------------------------------------------------------------------
+
+header() {
+    echo "----------------------------------------"
+    echo $1
+}
+
+footer() {
+    echo ""
+}
+
+run_on_all_cores() {
+    for (( core=0; core<NUM_CORES; core++ ))
+    do
+        if [[ $2 == "true" ]]; then
+            echo -e "$CC""core:$CB #$core$CE"
+            sudo env "PATH=$PATH" bfm vmcall string json "$1"
+            echo -e ""
+        else
+            sudo env "PATH=$PATH" bfm vmcall string json "$1" > /dev/null
+        fi
+    done
+}
+
+# ------------------------------------------------------------------------------
+# Init
+# ------------------------------------------------------------------------------
+
+run_on_all_cores "{\"command\":\"clear_cpuid_access_log\"}"
+run_on_all_cores "{\"command\":\"log_cpuid_access\", \"enabled\": true}"
+run_on_all_cores "{\"command\":\"reset_cpuid_all\"}"
+
+# ------------------------------------------------------------------------------
+# Tests
+# ------------------------------------------------------------------------------
+
+
+header "emulate cpuid"
+run_on_all_cores "{\"command\":\"emulate_cpuid\", \"leaf\": 0, \"subleaf\": 0,
+                    \"eax\": \"00000000000000000000000000001000\",
+                    \"ebx\": \"00000000000000000000000000001000\",
+                    \"ecx\": \"00000000000000000000000000001000\",
+                    \"edx\": \"00000000000000000000000000001000\"}"
+run_on_all_cores "{\"command\":\"emulate_cpuid\", \"leaf\": 1, \"subleaf\": 0,
+                    \"eax\": \"00000000000000000000000000000001\",
+                    \"ebx\": \"00000000000000000000000000000111\",
+                    \"ecx\": \"00000000000000000000000000000011\",
+                    \"edx\": \"00000000000000000000000000001000\"}"
+run_on_all_cores "{\"command\":\"emulate_cpuid\", \"leaf\": 2, \"subleaf\": 0,
```

Comment:
missing tests for providing invalid garbage strings, and a mix of "0, 1, and -".